Middleware-with-Tor/tor/tor/middlewares.py

```diff
@@ -0,0 +1,61 @@
+import sys
+import os
+from dotenv import load_dotenv
+from stem import Signal
+from stem.control import Controller
+from scrapy.downloadermiddlewares.httpproxy import HttpProxyMiddleware
+from stem.util.log import get_logger
+
+sys.path.append(os.path.abspath(".."))
+
+from settings import REQUESTS_PER_SAME_TOR_IDENTITY
```

Comment:
You should access settings through Scrapy framework, not directly import it
See https://docs.scrapy.org/en/latest/topics/settings.html#how-to-access-settings

Reply:
Fixed.